planner/core/logical_plan_builder.go

```diff
@@ -1901,6 +1901,38 @@ func (b *PlanBuilder) buildDataSource(tn *ast.TableName) (LogicalPlan, error) {
 
 	tableInfo := tbl.Meta()
 	b.visitInfo = appendVisitInfo(b.visitInfo, mysql.SelectPriv, dbName.L, tableInfo.Name.L, "")
+	charset, collation := b.ctx.GetSessionVars().GetCharsetInfo()
+	if tableInfo.IsView() {
+		var (
+			selectNode        ast.StmtNode
+			selectLogicalPlan Plan
+		)
+		selectNode, err = parser.New().ParseOneStmt(tableInfo.View.SelectStmt, charset, collation)
+		if err != nil {
+			return nil, err
+		}
+		selectLogicalPlan, err = b.Build(selectNode)
+		if err != nil {
+			return nil, err
+		}
+		var viewCols []*expression.Column
+		var selectSchema = selectLogicalPlan.Schema()
+		if len(selectSchema.Columns) != len(tableInfo.View.Cols) {
+			return nil, errors.New(fmt.Sprintf("View %s's UnderTable has changed with different column count", tableInfo.Name.L))
+		}
+		for i := range tableInfo.View.Cols {
+			if selectSchema.Columns[i].ColName != tableInfo.View.Cols[i] {
```

Comment:
This is not correct.
We should not return an error if the column order is changed of the underlying table.

We can use schema.FindColumnByName here.